vmm/src/vm.rs

```diff
@@ -513,11 +516,22 @@ impl DeviceManager {
             Box::new(KernelIoapicIrq::new(serial_evt))
         };
 
-        // Add serial device
-        let serial = Arc::new(Mutex::new(devices::legacy::Serial::new_out(
-            interrupt,
-            Box::new(stdout()),
-        )));
+        let serial = if vm_cfg.serial.mode != SerialOutputMode::Off {
+            let writer: Box<io::Write + Send> = match vm_cfg.serial.mode {
+                SerialOutputMode::File => Box::new(
+                    File::create(vm_cfg.serial.file.unwrap())
+                        .map_err(DeviceManagerError::SerialOutputFileOpen)?,
+                ),
+                _ => Box::new(stdout()),
```

Comment:
This is wildcard to also match "Off" which we've already checked for earlier but will otherwise result in a compile failure as it won't be matched. However using an extra Option<> really makes things less readable.

Reply:
Refactored this code so this statement doesn't apply any more.